gems/smithy-schema/lib/smithy-schema/shapes.rb

```diff
@@ -30,28 +30,52 @@ def []=(key, value)
         end
       end
 
+      # A reference to a shape.
+      class ShapeRef
+        def initialize(options = {})
+          @target = options[:target]
+          @location = options[:location]
```

Comment:
I'm hoping that we stay away from `location` name since it makes sense with the context of C2j models but to me - I can't see that obvious correlation that it is the represented `memberName` part of a "parent" shape (when using to represent member shapes)

Reply:
Talked offline, I think location_name makes sense. Location is the protocol specific bit that we don't need anymore.